### PR TITLE
DAT-3993 | Change template title while editing infobox template

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/DAT-3993

mercury part: https://github.com/Wikia/mercury/pull/2279

### CODE REVIEW

File: extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php [hunk 1]
```diff
@@ -39,20 +39,46 @@ public function getAssets() {
...
 
+		$response = $this->getResponse();
+		$response->setFormat( WikiaResponse::FORMAT_JSON );
 		$response->setVal( 'urls', $urls );
 		$response->setVal( 'success', $status->isOK() );
 		$response->setVal( 'errors', $status->getErrorsArray() );
 		$response->setVal( 'warnings', $status->getWarningsArray() );
 	}
 
+	public function publishAndRename() {
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php]: I know we didn't do that in the publish() method, yet I think it's high time to describe the expected request params to make sure it's clear what we're receiving.

File: extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php [hunk 2]
```diff
@@ -39,20 +39,46 @@ public function getAssets() {
...
+
+		// if result of moveTo operation is not true, extend array with errors
+		if ( $moveToResult !== true ) {
+			$errors = array_merge( $errors, $moveToResult );
+			$success = false;
+		}
+
+		$response = $this->getResponse();
+		$response->setFormat( WikiaResponse::FORMAT_JSON );
+		$response->setVal( 'urls', $urls );
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php]: Do I read correctly that if move is unsuccessful, we will provide url to 'new' title anyway ( https://github.com/Wikia/app/pull/10144/files#diff-8cddc7012469fc8c05dfd56e7a4f14b6R60 )?

File: extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php [hunk 3]
```diff
@@ -39,15 +39,33 @@ public function getAssets() {
...
+	public function publishAndRename() {
 		$requestParams = $this->getRequest()->getParams();
-		$urls = PortableInfoboxBuilderHelper::createRedirectUrls( $requestParams['title'] );
+		$status = $this->attemptSave( $requestParams );
 
+		$newTitleString = $requestParams['newTitle'];
+		$urls = PortableInfoboxBuilderHelper::createRedirectUrls( $newTitleString );
+		$oldTitle = PortableInfoboxBuilderHelper::getTitle( $requestParams['title'], $status );
+		$newTitle = PortableInfoboxBuilderHelper::getTitle( $newTitleString, $status );
+		$moveToResult = $oldTitle->moveTo($newTitle, false, "Infobox Builder: rename infobox template");
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php]: This is something that we need to internationalize (the move message)

**Comment by RafLeszczynski** [comment]
LGTM

**Comment by RafLeszczynski** [comment]
LGTM

**Pull request merged**